homeassistant/components/sma/__init__.py

```diff
@@ -197,6 +200,18 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok:
         data = hass.data[DOMAIN].pop(entry.entry_id)
         await data[PYSMA_OBJECT].close_session()
-        data[PYSMA_REMOVE_LISTENER]()
+        for remove_listener in data[PYSMA_REMOVE_LISTENER]:
+            remove_listener()
 
     return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    interval = timedelta(
+        seconds=entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
```

Comment:
We want to avoid exposing a scan or polling interval, instead we should provide sane defaults.

Please explain why this is required to add (or problematic not to have)